### PR TITLE
initial design

### PR DESCRIPTION
**Pull request by MatheMax**
It is now possible to get a heuristic initial design based on the characteristics of a one-stage design for a z-test. 

The only new function is 'get_initial_design'

It is accompanied by documentation and a test.

### CODE REVIEW

File: R/minimize.R [hunk 1]
```diff
@@ -115,6 +115,48 @@ print.adoptrOptimizationResult <- function(x, ...) {
...
+#'    type  = "two-stage",
+#'    order = 7L
+#' )
+#'
+#' @export
+get_initial_design <- function(theta, alpha = 0.025, beta = 0.2,
+                               type = c("two-stage", "group-sequential", "one-stage"),
+                               order = 7L) {
+    type <- match.arg(type)
+    if(alpha <= 0 || alpha >= 1 || beta <= 0 || beta >= 1)
```
Comment (kkmann) [inline R/minimize.R]: please try to follow common R style patterns, i.e. space after if (I know it's a function, but that makes code a lot easier to read) cf. https://style.tidyverse.org/syntax.html#parentheses

File: R/minimize.R [hunk 2]
```diff
@@ -115,6 +115,48 @@ print.adoptrOptimizationResult <- function(x, ...) {
...
+        stop("alpha and beta must be in (0, 1)!")
+    c    <- stats::qnorm(1 - alpha)
+    n    <- 2 * (c + stats::qnorm(1 - beta))^2 / theta^2
```
Comment (kkmann) [inline R/minimize.R]: doesn't that depend on how many arms we have?

**Comment by MatheMax** [comment]
@kkmann it depends on the DataDistribution now :)

File: R/minimize.R [hunk 3]
```diff
@@ -115,6 +115,57 @@ print.adoptrOptimizationResult <- function(x, ...) {
...
+#' )
+#'
+#' @export
+get_initial_design <- function(theta, alpha = 0.025, beta = 0.2,
+                               type = c("two-stage", "group-sequential", "one-stage"),
+                               dist = Normal(), order = 7L, ...) {
+    type <- match.arg(type)
+    if (alpha <= 0 || alpha >= 1 || beta <= 0 || beta >= 1)
+        stop("alpha and beta must be in (0, 1)!")
+    theta <- ifelse(dist@two_armed, theta / sqrt(2), theta)
```
Comment (kkmann) [inline R/minimize.R]: doesnt that depend on the data distribution as well?

**Comment by kkmann** [comment]
@MatheMax I trust you on this one, your call!

**Comment by kkmann** [comment]
@MatheMax sorry, but you still need to update the vignettes to use the new function (+ minimal explanation of what it is doing)

**Comment by kkmann** [comment]
looking good now, I can't think of anything else ;)

**Pull request merged**